atlas-sdk/src/main/java/com/atlas/sdk/AtlasSdk.kt

```diff
@@ -330,29 +341,34 @@ object AtlasSdk {
         return requireStatsUpdate
     }
 
-    private suspend fun login(user: AtlasUser?): AtlasUser? {
-        user?.let { _atlasUser ->
-            if (_atlasUser.id.isNotEmpty() && _atlasUser.atlasId.isNullOrEmpty()) {
-                val response = userRemoteRepository.login(appId, _atlasUser)
-                if (response != null && response.isSuccessful) {
-                    // we may have OR may not have atlasId in the beginning
-                    // but now we should have it in response
-                    return _atlasUser.copy(atlasId = response.id)
+    private fun login(user: AtlasUser?): CompletableFuture<AtlasUser?> {
+        return CompletableFuture.supplyAsync() {
+            user?.let { _atlasUser ->
+                if (_atlasUser.id.isNotEmpty() && _atlasUser.atlasId.isNullOrEmpty()) {
+                    val response = userRemoteRepository.login(appId, _atlasUser)
+                    if (response != null && response.get()?.isSuccessful == true) {
+                        // we may have OR may not have atlasId in the beginning
+                        // but now we should have it in response
+                        return@supplyAsync _atlasUser.copy(atlasId = response.get()?.id)
+                    }
+                } else {
+                    return@supplyAsync _atlasUser
                 }
-            } else {
-                return _atlasUser
             }
+            return@supplyAsync null
         }
-        return null
     }
 
-    private suspend fun fetchConversations(atlasUser: AtlasUser): List<ConversationStats>? {
-        val response = conversationsRemoteRepository.fetchConversations(atlasUser)
-        if (response != null && response.isSuccessful) {
-            return response.data?.map { ConversationStats.fromConversation(it) }
+    private fun fetchConversations(atlasUser: AtlasUser): CompletableFuture<List<ConversationStats>?> {
+        return CompletableFuture.supplyAsync {
+            val response = conversationsRemoteRepository.fetchConversations(atlasUser)
+            if (response.get() != null && response.get()?.isSuccessful == true) {
+                return@supplyAsync response.get()?.data?.map { ConversationStats.fromConversation(it) }
+            }
+            return@supplyAsync null
         }
-        return null
-    }
+        }
```

Comment:
I was questioning AI about avoiding the call of `.get()` and the answer I have got says that using coroutines is more idiomatic to Kotlin. What do you think about it?

![CleanShot 2024-11-14 at 11 47 02@2x](https://github.com/user-attachments/assets/aab62308-8e6a-48b9-ab07-a2c80d9ac9bf)

Reply:
How about 
```
    fun identify(userId: String? = null, userHash: String? = null, userName: String? = null, userEmail: String? = null): CompletableFuture<Void> {
        return CompletableFuture.runAsync {
            val user = AtlasUser(
                userId ?: "",
                userHash ?: "",
                null,
                userName,
                userEmail
            )
            restore(user)
        }
    }.get()
```

Reply:
- `identify()`
- `updateCustomFields()`